internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -751,7 +756,12 @@ func (r *Reconciler) updateMachineDeployment(ctx context.Context, s *scope.Scope
 		return nil
 	}
 
-	log.Infof("Patching %s", tlog.KObj{Obj: currentMD.Object})
+	changes := patchHelper.Changes()
+	if len(changes) == 0 {
+		log.Infof("Patching %s", tlog.KObj{Obj: currentMD.Object})
+	} else {
+		log.Infof("Patching %s, diff: %s", tlog.KObj{Obj: currentMD.Object}, changes)
+	}
```

Comment:
What if we merge with the if above
```suggestion
	if !patchHelper.HasChanges() {
		log.V(3).Infof("No changes for %s", tlog.KObj{Obj: currentMD.Object})
		return nil
	} else {
		log.Infof("Patching %s, diff: %s", tlog.KObj{Obj: currentMD.Object}, patchHelper.Changes())
	}
```

Reply:
I think I would still like to handle both cases (changes being empty or not). We have cases where HasChanges returns true, but Changes are empty. Technically this should only ever happen on create calls.

But to be honest I think it's brittle to depend on this exact behavior of the patch helper today. I don't want our logs to degrade if we introduce another case in the patch helper where HasChanges is true, but Changes returns an empty byte array